static/js/reactions.js

```diff
@@ -497,4 +520,62 @@ export function update_user_fields(clean_reaction_object) {
     } else {
         clean_reaction_object.class = "message_reaction";
     }
+
+    clean_reaction_object.count = clean_reaction_object.user_ids.length;
+    clean_reaction_object.label = generate_title(
+        clean_reaction_object.emoji_name,
+        clean_reaction_object.user_ids,
+    );
+
+    // The vote_text field set here is used directly in the Handlebars
+    // template for rendering (or rerendering!) a message.
+    clean_reaction_object.vote_text = get_vote_text(clean_reaction_object, should_display_reactors);
+}
+
+export function get_vote_text(clean_reaction_object, should_display_reactors) {
+    if (should_display_reactors) {
+        return comma_separated_usernames(clean_reaction_object.user_ids);
+    }
+    return `${clean_reaction_object.user_ids.length}`;
+}
+
+function check_should_display_reactors(cleaned_reactions) {
+    if (!user_settings.display_emoji_reaction_users) {
+        return false;
+    }
+
+    let total_reactions = 0;
+    for (const r of cleaned_reactions.values()) {
+        total_reactions += r.count;
+    }
+    return total_reactions <= 3;
+}
+
+function comma_separated_usernames(user_list) {
+    const usernames = people.get_display_full_names(user_list);
+    const current_user_has_reacted = user_list.includes(page_params.user_id);
+
+    if (current_user_has_reacted) {
+        const current_user_index = user_list.indexOf(page_params.user_id);
+        usernames[current_user_index] = $t({
+            defaultMessage: "You",
+        });
+    }
+    const comma_separated_usernames = usernames.join(", ");
+    return comma_separated_usernames;
+}
+
+export function update_vote_text_on_message(message) {
+    // Because whether we display a count or the names of reacting
+    // users depends on total reactions on the message, we need to
+    // recalculate this whenever adjusting reaction rendering on a
+    // message.
+    const cleaned_reactions = get_message_reactions(message);
+    const should_display_reactors = check_should_display_reactors(cleaned_reactions);
+    for (const [reaction, clean_reaction] of message.clean_reactions.entries()) {
+        const reaction_elem = find_reaction(message.id, clean_reaction.local_id);
+        const vote_text = get_vote_text(clean_reaction, should_display_reactors);
+        message.clean_reactions.get(reaction).vote_text = vote_text;
+        set_reaction_vote_text(reaction_elem, vote_text);
+    }
```

Comment:
I changed this to update `message.clean_reactions` too for `vote_text`. I don't think it fixes a bug, but felt like the correct thing to do.